pkg/identity/hsmsign_test.go

```diff
@@ -0,0 +1,134 @@
+			return libraryLocation
+		}
+	}
+	return "NoLibraryFound"
+}
+
+func TestNewHSMSignerFactory(t *testing.T) {
+
+	t.Run("No library provided", func(t *testing.T) {
+		_, err := NewHSMSignerFactory("")
+		require.EqualError(t, err, "library path not provided", "Expected Error message about no library provided")
+	})
+
+	t.Run("No library found", func(t *testing.T) {
+		_, err := NewHSMSignerFactory("/some/path/notalibrary.so")
+		require.EqualError(t, err, "instantiation failed for /some/path/notalibrary.so", "Expected Error message about bad library")
+	})
+
+	t.Run("Library found", func(t *testing.T) {
+
+		// SoftHSM must be installed for this to pass
+		hsmSignerFactory, err := NewHSMSignerFactory(findSoftHSMLibrary())
+		if err != nil {
+			t.Fatal(err)
+		}
+		hsmSignerFactory.Dispose()
+	})
+}
+
+func TestNewHSMSigner(t *testing.T) {
+
+	// SoftHSM must be installed for this to pass
+	hsmSignerFactory, err := NewHSMSignerFactory(findSoftHSMLibrary())
+	require.NoError(t, err)
+	defer hsmSignerFactory.Dispose()
+
+	t.Run("No options provided", func(t *testing.T) {
+		hsmSignerOptions := HSMSignerOptions{}
+		_, _, err := hsmSignerFactory.NewHSMSigner(hsmSignerOptions)
+		require.EqualError(t, err, "no Label provided", "Expected Error message about label")
+	})
+
+	t.Run("No Label provided", func(t *testing.T) {
+		hsmSignerOptions := HSMSignerOptions{
+			Pin:        "12345",
+			Identifier: "Fred",
+		}
+		_, _, err := hsmSignerFactory.NewHSMSigner(hsmSignerOptions)
+		require.EqualError(t, err, "no Label provided", "Expected Error message about label")
+	})
+
+	t.Run("No Pin provided", func(t *testing.T) {
+		hsmSignerOptions := HSMSignerOptions{
+			Label:      "12345",
+			Identifier: "Fred",
+		}
+		_, _, err := hsmSignerFactory.NewHSMSigner(hsmSignerOptions)
+		require.EqualError(t, err, "no Pin provided", "Expected Error message about pin")
+	})
+
+	t.Run("No Identifier provided", func(t *testing.T) {
+		hsmSignerOptions := HSMSignerOptions{
+			Pin:   "12345",
+			Label: "Fred",
+		}
+		_, _, err := hsmSignerFactory.NewHSMSigner(hsmSignerOptions)
+		require.EqualError(t, err, "no Identifier provided", "Expected Error message about identifier")
+	})
+
+	t.Run("Label does not exist", func(t *testing.T) {
+		hsmSignerOptions := HSMSignerOptions{
+			Pin:        "98765432",
+			Label:      "NoTokenWithThisLabel",
+			Identifier: "fred",
+		}
+		_, _, err := hsmSignerFactory.NewHSMSigner(hsmSignerOptions)
+		require.EqualError(t, err, "could not find token with label NoTokenWithThisLabel", "Expected Error message about label")
+	})
+
+	t.Run("Pin Incorrect", func(t *testing.T) {
+
+		// Token with "ForFabric" Label must have been created for this test to pass
+		hsmSignerOptions := HSMSignerOptions{
+			Pin:        "911",
+			Label:      "ForFabric",
+			Identifier: "fred",
+		}
+		_, _, err := hsmSignerFactory.NewHSMSigner(hsmSignerOptions)
+		require.Containsf(t, err.Error(), "CKR_PIN_INCORRECT", "Expected Error message invalid Pin")
```

Comment:
I think this will panic if no error is thrown. You might need a `require.Error(err)` assertion before the `require.Contains()`